src/test/java/com/icthh/xm/ms/entity/web/rest/FunctionResourceMvcIntTest.java

```diff
@@ -78,6 +83,70 @@ public void testCallPostAnonymousFunction() {
         verify(functionService).executeAnonymous(eq("SOME-ANONYMOUS-FUNCTION_KEY.TROLOLO"), eq(of("var1", "val1", "var2", "val2")), eq("POST"));
     }
 
+    @Test
+    @SneakyThrows
+    public void testCallPostAnonymousFunctionWithUrlEncodedParams() {
+        when(functionService.executeAnonymous("SOME-ANONYMOUS-FUNCTION_KEY.TROLOLO", of("var1", "val1", "var2", "val2"), "POST"))
+            .thenReturn(new FunctionContext().data(of("test", "result")));
+        mockMvc.perform(post("/api/functions/anonymous/SOME-ANONYMOUS-FUNCTION_KEY.TROLOLO")
+                .contentType(MediaType.APPLICATION_FORM_URLENCODED)
+                .param("var1", "val1")
+                .param("var2", "val2")
+            )
+            .andDo(print())
+            .andExpect(jsonPath("$.data.test").value("result"))
+            .andExpect(status().isCreated());
+        verify(functionService).executeAnonymous(eq("SOME-ANONYMOUS-FUNCTION_KEY.TROLOLO"), eq(of("var1", "val1", "var2", "val2")), eq("POST"));
+    }
+
+    @Test
+    @SneakyThrows
+    public void testCallPostAnonymousFunctionWithPathWithUrlEncodedParams() {
+        when(functionService.executeAnonymous("SOME-ANONYMOUS-FUNCTION_KEY/TROLOLO", of("var1", "val1", "var2", "val2"), "POST"))
+            .thenReturn(new FunctionContext().data(of("test", "result")));
+        mockMvc.perform(post("/api/functions/anonymous/SOME-ANONYMOUS-FUNCTION_KEY/TROLOLO")
+                .contentType(MediaType.APPLICATION_FORM_URLENCODED)
+                .param("var1", "val1")
+                .param("var2", "val2")
+            )
+            .andDo(print())
+            .andExpect(jsonPath("$.data.test").value("result"))
+            .andExpect(status().isCreated());
+        verify(functionService).executeAnonymous(eq("SOME-ANONYMOUS-FUNCTION_KEY/TROLOLO"), eq(of("var1", "val1", "var2", "val2")), eq("POST"));
+    }
+
+    @Test
+    @SneakyThrows
+    public void testCallPostFunctionWithUrlEncodedParams() {
+        when(functionService.execute("SOME-ANONYMOUS-FUNCTION_KEY.TROLOLO", of("var1", "val1", "var2", "val2"), "POST"))
```

Comment:
In this case, functionKey SOME-FUNCTION_KEY.TROLOLO instead of SOME-ANONYMOUS-FUNCTION_KEY.TROLOLO would be more descriptive